protocol/carbon/carbon_test.go

```diff
@@ -21,6 +21,10 @@ func TestNewCarbonDatapoint(t *testing.T) {
 	assert.Equal(t, nil, err, "Should be a valid carbon line")
 	assert.Equal(t, "hello", dp.Metric, "Should get metric back")
 
+	floatDp, err := NewCarbonDatapoint("hello 3 1519398226.544148", identityParser)
+	assert.Equal(t, nil, err, "Should be a valid carbon line")
+	assert.Equal(t, "hello", floatDp.Metric, "Should get floating timestamp metric back")
```

Comment:
check the timestamp here as well please.  This looks like  a great opportunity to make this a table driven test.